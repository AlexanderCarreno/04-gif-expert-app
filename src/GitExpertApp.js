//rafc
import React, { useState } from 'react';

const GitExpertApp = ({}) => {
    //const  categories = ['One Punch', 'Guerra de magos', 'Demon Slayer'];

    const [categories, setCategories] = useState(['One Punch', 'Guerra de magos', 'Demon Slayer']);

    const handleAdd = () => {
        //const newElement = ['My Hero Academia'];
        //setCategories([...categories, ...newElement]);

        //setCategories([...categories, 'My Hero Academia']);

        setCategories( cats => [...cats, 'My Hero Academia'])
    }


    return(
        <>
            <h2>GitExpertApp</h2>
            <hr/>

            <button onClick={ handleAdd }>Agregar</button>

            <ol>
                { 
                    categories.map( category =>{
                        return <li key={ category } >{category}</li>
                    })
                }
            </ol>
        </>
    )
}

export default GitExpertApp;