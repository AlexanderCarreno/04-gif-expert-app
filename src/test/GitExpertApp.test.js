import React from 'react';
import { shallow } from 'enzyme';
import GitExpertApp from '../GitExpertApp';

describe('debe mostrar el componente correctamente', () => { 
    
    let wrapper = shallow( <GitExpertApp />);

    beforeEach( ()=>{
        wrapper = shallow( <GitExpertApp />);
    });

    test('debe mostrar <GitExpertApp /> corractamente', () => { 
        expect( wrapper ).toMatchSnapshot();
     })
});