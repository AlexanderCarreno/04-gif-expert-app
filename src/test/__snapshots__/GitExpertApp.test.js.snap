// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe mostrar el componente correctamente debe mostrar <GitExpertApp /> corractamente 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="My Hero Academia"
      key="My Hero Academia"
    />
  </ol>
</Fragment>
`;
